src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -430,9 +482,10 @@ class RemoteMachineTrainingService implements TrainingService {
             } else {
                 deferred.reject(new Error(`No valid passwd or sshKeyPath is configed.`));
             }
-            this.machineSSHClientMap.set(rmMeta, conn);
+            
             conn.on('ready', async () => {
-                this.machineSSHClientMap.set(rmMeta, conn);
+                //Initialize connection, the first connection is used for gpu collector
```

Comment:
Consider to move all the ssh connection related code into SSHClientManager

Reply:
updated.